src/contracts/middleware/IndexRegistry.sol

```diff
@@ -207,18 +214,44 @@ contract IndexRegistry is IIndexRegistry {
         _operatorIdToIndexHistory[operatorId][quorumNumber][_operatorIdToIndexHistory[operatorId][quorumNumber].length - 1].toBlockNumber = uint32(block.number);
     }
 
-    /**
-     * @notice remove an operator from the globalOperatorList  
-     * @param indexOfOperatorToRemove index of the operator to remove
-     */ 
-    function _removeOperatorFromGlobalOperatorList(uint32 indexOfOperatorToRemove) internal {
-        uint32 globalOperatorListLastIndex = uint32(globalOperatorList.length - 1);
-        bytes32 operatorIdToSwap;
-        if(indexOfOperatorToRemove != globalOperatorListLastIndex){
-            operatorIdToSwap = globalOperatorList[globalOperatorListLastIndex];
-            globalOperatorList[indexOfOperatorToRemove] = operatorIdToSwap;
-            emit GlobalIndexUpdate(operatorIdToSwap, indexOfOperatorToRemove);
+    /// @notice Returns the total number of operators of the service for the given `quorumNumber` at the given `blockNumber`
+    function _getTotalOperatorsForQuorumAtBlockNumber(uint8 quorumNumber, uint32 blockNumber) internal view returns (uint32){
+        // if there are no entries in the total operator history, return 0
+        if (_totalOperatorsHistory[quorumNumber].length == 0) {
+            return 0;
         }
-        globalOperatorList.pop();
+        
+        // if there is only one entry in the total operator history, return it
+        if (_totalOperatorsHistory[quorumNumber].length == 1) {
+            return _totalOperatorsHistory[quorumNumber][_totalOperatorsHistory[quorumNumber].length - 1].index;
+        }
+
+        // loop backwards through the total operator history to find the total number of operators at the given block number
+        for (uint i = _totalOperatorsHistory[quorumNumber].length - 2; i >= 0; i--) {
```

Comment:
possible bug: shouldn't we still be starting at `_totalOperatorsHistory[quorumNumber].length - 1`? why skip the last entry? what if it is the correct one (and there are 2 or more entries in the history)?

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/81/commits/7d71e880e8adaaeca4a5b8a919abcbb852610fa7 ah, i think this fixes?

Reply:
can you check (i.e. _write a test_) that the loop doesn't fail for trying to underflow when the correct entry is the 0-indexed one?
I previously had some interesting underflow behavior w/ Solidity when using decrementing loops.

Also, it seems like right now this will return `1` (the value of `_totalOperatorsHistory[quorumNumber][0].index`, right?) even if the provided `blockNumber` is before the first operator joined.

Reply:
oh, also going to point out that this implementation would return `_totalOperatorsHistory[quorumNumber].length - 1` for an input value of `blockNumber` that is in the future, but I guess that's just a general behavior with these types of function / doesn't seem like a particular problem?
Maybe something to note with an `@dev` tag though.

Reply:
i'm trying to push off writing tests to a future pr. this functionality more needs to pass out mental test to get merged into multiquourms, but needs to severely tested before master

Reply:
OK, fair enough -- I'm OK putting off the serious tests until later.
I still think this is somewhat broken for at least one edge case, and am ~50% sure it will break with an underflow revert in the loop (and that `i > 0` is the correct condition).

Reply:
ahh, i think you're right. here: 1e35bd8d